bundle/config/mutator/run_as.go

```diff
@@ -53,14 +53,17 @@ func (e errBothSpAndUserSpecified) Error() string {
 }
 
 func validateRunAs(b *bundle.Bundle) error {
-	runAs := b.Config.RunAs
-
-	// Error if neither service_principal_name nor user_name are specified
-	if runAs.ServicePrincipalName == "" && runAs.UserName == "" {
+	// Error if neither service_principal_name nor user_name are specified, but the
+	// run_as section is present.
+	// TODO: Track location for nil values. This is likely to be a separate PR.
+	// TODO: Change comparision from directly comparing the dyn.Values to
+	// checking the kind instead.``
```

Comment:
The location of nils is already tracked. More likely we're returning `dyn.NilValue` somewhere after a nil check instead of returning the original value.

Reply:
This was fixed in https://github.com/databricks/cli/pull/1523